src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -364,14 +369,18 @@ public function getCompiler()
      */
     public function set($id, $service)
     {
-        $id = strtolower($id);
+        $caseSensitiveId = (string) $id;
```

Comment:
Probably to be BC with the previous behavoir: It was possible to pass e.g. an object with `__toString()` as `$id` as `strtolower()` cast everything to a string.